presto-hive/src/main/java/io/prestosql/plugin/hive/procedure/SyncPartitionMetadataProcedure.java

```diff
@@ -144,28 +145,31 @@ private void doSyncPartitionMetadata(ConnectorSession session, String schemaName
         syncPartitions(partitionsToAdd, partitionsToDrop, syncMode, metastore, session, table);
     }
 
-    private static List<FileStatus> listDirectory(FileSystem fileSystem, FileStatus current, List<Column> partitionColumns, int depth)
+    private static List<FileStatus> listDirectory(ConnectorSession session, FileSystem fileSystem, FileStatus current, List<Column> partitionColumns, int depth)
     {
         if (depth == 0) {
             return ImmutableList.of(current);
         }
 
         try {
             return Stream.of(fileSystem.listStatus(current.getPath()))
-                    .filter(fileStatus -> isValidPartitionPath(fileStatus, partitionColumns.get(partitionColumns.size() - depth)))
-                    .flatMap(directory -> listDirectory(fileSystem, directory, partitionColumns, depth - 1).stream())
+                    .filter(fileStatus -> isValidPartitionPath(session, fileStatus, partitionColumns.get(partitionColumns.size() - depth)))
+                    .flatMap(directory -> listDirectory(session, fileSystem, directory, partitionColumns, depth - 1).stream())
                     .collect(toImmutableList());
         }
         catch (IOException e) {
             throw new PrestoException(HIVE_FILESYSTEM_ERROR, e);
         }
     }
 
-    private static boolean isValidPartitionPath(FileStatus file, Column column)
+    private static boolean isValidPartitionPath(ConnectorSession session, FileStatus file, Column column)
     {
-        Path path = file.getPath();
+        String path = file.getPath().getName();
+        if (isAllowMixedCasePartitions(session)) {
```

Comment:
it should rather be procedure (optional) parameter. Sorry for miscommunication on this.

Reply:
My bad, I misread the Github issue description.